cirq-core/cirq/sim/act_on_state_vector_args.py

```diff
@@ -213,6 +213,9 @@ def _strat_act_on_state_vector_from_mixture(action: Any, args: 'cirq.ActOnStateV
     unitary = unitaries[index].astype(args.target_tensor.dtype).reshape(shape)
     linalg.targeted_left_multiply(unitary, args.target_tensor, args.axes, out=args.available_buffer)
     args.swap_target_tensor_for(args.available_buffer)
+    if protocols.is_measurement(action):
```

Comment:
I've found `protocols.is_measurement` is extremely slow, so this could have a perf impact if this is called frequently. (I couldn't use it in `_core_iterator` without a big perf regression).

Reply:
Thanks for calling that out! I'll avoid it here, but I'm curious about what makes it slow - do you recall the general structure of the circuits you saw this regression on? I can see this calls `measurement_keys`, but that ought to be quick for everything except CircuitOperations or similarly-nested structures.

Reply:
I didn't deep dive since I needed specifically only to decouple after single-gate measurements, so is_measurement was the wrong condition there anyway. Though I also remember being surprised that the perf was so impacted.

Reply:
I've figured out the reason for the slowness. Let me send out a fix soon.

Reply:
#4140